frontend/frontend/protocols/ssh/_proxy_handler.py

```diff
@@ -0,0 +1,324 @@
+        :param pixelwidth: The width of the screen in pixels, if known (may be 0 if unknown)
+        :param pixelheight: The height of the screen in pixels, if known (may be 0 if unknown)
+        :return: True if the pty request succeeded
+        """
+        try:
+            backend_channel = self._get_corresponding_backend_channel(attacker_channel.chanid)
+            backend_channel.get_pty(term_string, width, height, pixelwidth, pixelheight)
+        except KeyError:
+            return False
+        except SSHException:
+            logger.error("Failed to get pty on the backend")
+            return False
+
+        logger.info("Attacker sent pty request on channel %s", attacker_channel.chanid)
+        return True
+
+    def handle_window_change_request(self, attacker_channel: Channel, width: int, height: int,
+                                     pixelwidth: int, pixelheight: int) -> bool:
+        """Tries to send a window change request to the corresponding backend channel.
+
+        :param attacker_channel: The attacker channel
+        :param width: The width of the screen in characters
+        :param height: The height of the screen in characters
+        :param pixelwidth: The width of the screen in pixels, if known (may be 0 if unknown)
+        :param pixelheight: The height of the screen in pixels, if known (may be 0 if unknown)
+        :return: True if the window change request succeeded
+        """
+        try:
+            backend_channel = self._get_corresponding_backend_channel(attacker_channel.chanid)
+            backend_channel.resize_pty(width, height, pixelwidth, pixelheight)
+        except KeyError:
+            return False
+        except SSHException:
+            logger.error("Failed to resize pty on the backend")
+            return False
+        return True
+
+
+def try_send_data(data, send_method) -> bool:
+    """Tries to send data and catch exceptions
+
+    :param data: The data to send
+    :param send_method: The send method
+    :return: True if suceeded to send data
+    """
+    try:
+        send_method(data)
+    except socket.timeout:
+        logger.warning("Timed out while trying to send data")
+        return False
+    except socket.error:
+        logger.warning("Failed data")
+        return False
+    return True
+
+
+def proxy_data(
+        attacker_channel: paramiko.Channel,
+        backend_channel: paramiko.Channel,
+        session_log: SSHSession):
+    """This will proxy data between two channels and log the data
+
+    :param attacker_channel: The attacker channel
+    :param backend_channel: The backend channeel
+    :param session_log: The SSH logging session
+    """
+    attacker_channel.settimeout(10)
+    backend_channel.settimeout(10)
+
+    command_parser = CommandParser()
+    # While the attacker channel is not closed and has not send eof
+    while not (attacker_channel.eof_received or attacker_channel.closed):
```

Comment:
This long while loop basically reads data from the attacker channel and proxies it over to the other proxy channel. Currently a 100ms delay in each loop iteration